src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonIEnumerableConverterFactory.cs

```diff
@@ -0,0 +1,239 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    /// <summary>
+    /// Converter factory for all IEnumerable types.
+    /// </summary>
+    internal class JsonIEnumerableConverterFactory : JsonConverterFactory
+    {
+        private static readonly JsonIDictionaryConverter<IDictionary> s_IDictionaryConverter = new JsonIDictionaryConverter<IDictionary>();
+        private static readonly JsonIEnumerableConverter<IEnumerable> s_IEnumerableConverter = new JsonIEnumerableConverter<IEnumerable>();
+        private static readonly JsonIListConverter<IList> s_IListConverter = new JsonIListConverter<IList>();
```

Comment:
coding style nit: name violations here.
`s_IDictionaryConverter` -> `s_iDictionaryConverter`
etc.

Reply:
Hmm since this is matching a type name, I'd prefer to keep the uppercase I for readability. However I'll change to `s_converterForXXX`.